### PR TITLE
122 the design of cards on social tab

### PR DESCRIPTION
**Pull request by NLMOTS**

### CODE REVIEW

File: frontend/src/app/modules/user-card/user-card.component.ts [hunk 1]
```diff
@@ -1,21 +1,58 @@
...
+
+                const fullCountry = countriesLib.whereAlpha2(country)?.country;
+
+                this.userLanguageFlag = this.countriesService.getCountriesList()
+                    .find((s: { name: string | undefined; }) => s.name === fullCountry)?.flag;
+            }
+        }
+    }
+
+    buttomGoToMessage() {
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/user-card/user-card.component.ts:50]: buttom -> button

File: frontend/src/app/modules/user-card/user-card.component.ts [hunk 2]
```diff
@@ -1,21 +1,58 @@
...
 
     ngOnInit(): void {
         this.user.tags = [...new Set(this.user.tags)];
+        this.userCountryFlag = this.countriesService.getCountriesList()
+            .find((s: { name: string | undefined; }) => s.name === this.user.country)?.flag;
+        this.setFlagByLanguage();
+    }
+
+    setFlagByLanguage() {
+        if (this.user.language !== undefined) {
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/user-card/user-card.component.ts:36]: Reverse if

File: frontend/src/app/modules/user-card/user-card.component.ts [hunk 3]
```diff
@@ -1,21 +1,58 @@
...
+            const languageCode = languagesLib.getCode(this.user.language);
+
+            if (languageCode) {
+                const country = countryCodes[languageCode];
+
+                const fullCountry = countriesLib.whereAlpha2(country)?.country;
```
[the remark this answers is not in the document]
Comment (NLMOTS) [reply frontend/src/app/modules/user-card/user-card.component.ts:42]: I was searching for solution, but language not always bind to country, so i found this file, and copyied, anyway, some languages doesn't have only one country

File: frontend/src/app/modules/user-card/user-card.component.ts [hunk 4]
```diff
@@ -1,21 +1,58 @@
...
 
     ngOnInit(): void {
         this.user.tags = [...new Set(this.user.tags)];
+        this.userCountryFlag = this.countriesService.getCountriesList()
+            .find((s: { name: string | undefined; }) => s.name === this.user.country)?.flag;
+        this.setFlagByLanguage();
+    }
+
+    setFlagByLanguage() {
+        if (this.user?.language) {
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/user-card/user-card.component.ts:36]: if (!this.user...)
   return;
/// Else do your staff

File: frontend/src/app/modules/user-card/user-card.module.ts [hunk 5]
```diff
@@ -1,15 +1,15 @@
 import { CommonModule } from '@angular/common';
-import { NgModule } from '@angular/core';
+import { CUSTOM_ELEMENTS_SCHEMA, NgModule, NO_ERRORS_SCHEMA } from '@angular/core';
 import { UserCardComponent } from '@modules/user-card/user-card.component';
 import { SharedModule } from '@shared/shared.module';
 
 @NgModule({
     declarations: [UserCardComponent],
     imports: [CommonModule, SharedModule],
+    schemas: [NO_ERRORS_SCHEMA,
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/user-card/user-card.module.ts:9]: Let's not use this...

File: frontend/src/app/modules/user-card/user-card.module.ts [hunk 6]
```diff
@@ -1,15 +1,15 @@
...
+        CUSTOM_ELEMENTS_SCHEMA],
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/user-card/user-card.module.ts:10]: And do you really need this?

File: frontend/src/app/modules/user-card/user-card.component.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Anatolii-Ivanov) [reply frontend/src/app/modules/user-card/user-card.component.ts:42]: Ok, let's leave it for now

**Review by Anatolii-Ivanov — DISMISSED** [review]

File: frontend/src/app/modules/user-card/mappingLanguagetoCountry.ts [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+    ab: 'GE', // Abkhazian
+    ae: 'AE', // Avestan
+    af: 'ZA', // Afrikaans
+    ak: 'GH', // Akan
+    am: 'ET', // Amharic
+    an: 'ES', // Aragonese
+    ar: 'SA', // Arabic
+    as: 'IN', // Assamese
+    av: 'RU', // Avaric
+    ay: 'BO', // Aymara
```
Comment (tatianahutii) [inline frontend/src/app/modules/user-card/mappingLanguagetoCountry.ts:16]: we should have this in the backend

Comment (tatianahutii) [reply frontend/src/app/modules/user-card/mappingLanguagetoCountry.ts:16]: and why do we need codes? I believe we already have countries in the backend

File: frontend/src/app/modules/user-card/user-card.component.ts [hunk 8]
```diff
@@ -22,6 +31,30 @@ export class UserCardComponent implements OnInit {
...
+
+        const languageCode = languagesLib.getCode(this.user.language);
+
+        if (languageCode) {
+            const country = countryCodes[languageCode];
+
+            const fullCountry = countriesLib.whereAlpha2(country)?.country;
+
+            this.userLanguageFlag = this.countriesService.getCountriesList()
+                .find((s: { name: string | undefined; }) => s.name === fullCountry)?.flag;
```
Comment (tatianahutii) [inline frontend/src/app/modules/user-card/user-card.component.ts:52]: duplications

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: frontend/src/app/modules/user-card/mappingLanguagetoCountry.ts (hunk 7 above)
Comment (NLMOTS) [reply frontend/src/app/modules/user-card/mappingLanguagetoCountry.ts]: In this task, i need to display country flag, only by language ,i didn't found any libraries where all languages are mapped to countries so i created a map for this, and it's only for visual, i don't know if we need this in backend

**Review by Anatolii-Ivanov — APPROVED** [review]

**Review by SidletskyiOlexandr — APPROVED** [review]

**Pull request merged**